### PR TITLE
[pollyTTS] Replace custom TTS cache with common TTS cache

### PR DESCRIPTION
**Pull request by dalgwen**
Uses the common TTS cache instead of a custom one.
No new features

Advantages
    use common configuration
    cache size configuration available
    better concurrency management
    simplifying polytts code.

Drawback
    lose parameter for time eviction policy (but in favor of a cache size eviction policy)

Relates to openhab/openhab-addons#15110

It'a draft because I didn't manage to test it, as I cannot resolve the bundle on my dev environment :
image
(If someone has an idea ?)

### CODE REVIEW

File: bundles/org.openhab.voice.pollytts/src/main/java/org/openhab/voice/pollytts/internal/PollyTTSService.java [hunk 1]
```diff
@@ -167,17 +160,15 @@ public AudioStream synthesize(String inText, Voice voice, AudioFormat requestedF
...
                 throw new TTSException("Could not read from PollyTTS service");
             }
             logger.debug("Audio Stream for '{}' in format {}", text, requestedFormat);
-            AudioStream audioStream = new PollyTTSAudioStream(cacheAudioFile, requestedFormat);
+            AudioStream audioStream = new PollyTTSAudioStream(pollyAudioStream, requestedFormat);
             return audioStream;
-        } catch (AudioException ex) {
-            throw new TTSException("Could not create AudioStream: " + ex.getMessage(), ex);
-        } catch (IOException ex) {
+        } catch (AmazonPollyException ex) {
```
Comment (lolodomo) [inline bundles/org.openhab.voice.pollytts/src/main/java/org/openhab/voice/pollytts/internal/PollyTTSService.java:171]: The javadoc of `getTextToSpeech` is mentioning `IOException` but not `AmazonPollyException`.

**Review by lolodomo — COMMENTED** [review]
I have just one comment

File: bundles/org.openhab.voice.pollytts/src/main/java/org/openhab/voice/pollytts/internal/PollyTTSService.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.voice.pollytts/src/main/java/org/openhab/voice/pollytts/internal/PollyTTSService.java:171]: @dalgwen : any feedback?

Comment (dalgwen) [reply bundles/org.openhab.voice.pollytts/src/main/java/org/openhab/voice/pollytts/internal/PollyTTSService.java:171]: OK, I fixed the javadoc.
The IOException is no longer possible as it was thrown by file related operation used by the custom cache.
The AmazonPollyException was not catched before (it is a RuntimeException) and I thought it was a good thing to catch it here.

But caution : I did not tested it. I still have the same resolution issue.

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
Let's merge this PR and we will adjust it before OH 4 is released in case we have wrong feedback.

**Pull request merged**